pkg/cloudprovider/types.go

```diff
@@ -109,6 +90,39 @@ func (i *InstanceType) Allocatable() v1.ResourceList {
 	return i.allocatable.DeepCopy()
 }
 
+func (its InstanceTypes) OrderByPrice(reqs scheduling.Requirements) InstanceTypes {
+	// Order instance types so that we get the cheapest instance types of the available offerings
+	sort.Slice(its, func(i, j int) bool {
+		iPrice := math.MaxFloat64
+		jPrice := math.MaxFloat64
+		if len(its[i].Offerings.Available().Requirements(reqs)) > 0 {
+			iPrice = its[i].Offerings.Available().Requirements(reqs).Cheapest().Price
+		}
+		if len(its[j].Offerings.Available().Requirements(reqs)) > 0 {
+			jPrice = its[j].Offerings.Available().Requirements(reqs).Cheapest().Price
+		}
+		if iPrice == jPrice {
+			return its[i].Name < its[j].Name
+		}
+		return iPrice < jPrice
+	})
+	return its
+}
+
+// Compatible returns the list of instanceTypes based on the supported capacityType and zones in the requirements
+func (its InstanceTypes) Compatible(requirements scheduling.Requirements) InstanceTypes {
+	var filteredInstanceTypes []*InstanceType
+	for _, instanceType := range its {
+		for _, offering := range instanceType.Offerings.Available() {
```

Comment:
```suggestion
		for _, offering := range instanceType.Offerings.Available().Requirements(requirements) {
```

Looks like you might be able to just use the Requirements method on the offerings (maybe we should change this to `Compatible` now to match) to get the set of offerings that are compatible. If that number is above 0, then you cause return this instance type.

Reply:
Makes sense. I'm for less code if there isn't a huge performance optimization here. I'd suspect that the benefit that we would get here would be marginal.

Reply:
Also, renamed the method as needed. But this is going to change both the aws-provider and core package which means not taking latest from any of the packages will break compatibility for the customers. I am assuming we are fine with this ?